content/en/docs/concepts/storage/persistent-volumes.md

```diff
@@ -403,6 +410,31 @@ In the past, the annotation `volume.beta.kubernetes.io/mount-options` was used i
 of the `mountOptions` attribute. This annotation is still working, however
 it will become fully deprecated in a future Kubernetes release.
 
+### Node Affinity
+
+{{< note >}}
+**Note:** You do not need to set this field. It is automatically populated for aws, gce and azure volume block types. You need to explicitly set this in case of local volumes. Most other volume types do not need this field.
+{{< /note >}}
+
+A PV can specify [node affinity](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#volumenodeaffinity-v1-core) to define constraints that limit what nodes this volume can be accessed from. Only the nodes that satisfy the constraints can access the volume.
+
+Node affinity is conceptually similar to [NodeSelector](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#nodeselector-v1-core).
+
+NodeSelector is specified as field `required` of field `nodeAffinity` in the PersistentVolumeSpec.
```

Comment:
Do we need all of these details about the node selector, or can we just point to the assign-pod-node page?

Reply:
the reason why i documented them here is, the fields of nodeAffinity and volumeNodeAffinity are different. And the assign-pod-node page describes only the node affinity for pods.

Reply:
The fields inside nodeAffinity are different, however the nodeSelector type is exactly the same, so should mirror the original documentation on it.  I prefer not to duplicate documentation on the same field because they will get out of sync with each other.